core/src/main/java/org/apache/gravitino/storage/relational/service/ModelVersionMetaService.java

```diff
@@ -0,0 +1,288 @@
+import org.apache.gravitino.storage.relational.po.ModelVersionAliasRelPO;
+import org.apache.gravitino.storage.relational.po.ModelVersionPO;
+import org.apache.gravitino.storage.relational.utils.ExceptionUtils;
+import org.apache.gravitino.storage.relational.utils.POConverters;
+import org.apache.gravitino.storage.relational.utils.SessionUtils;
+import org.apache.gravitino.utils.NameIdentifierUtil;
+import org.apache.gravitino.utils.NamespaceUtil;
+import org.glassfish.jersey.internal.guava.Lists;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ModelVersionMetaService {
+
+  private static final Logger LOG = LoggerFactory.getLogger(ModelVersionMetaService.class);
+
+  private static final ModelVersionMetaService INSTANCE = new ModelVersionMetaService();
+
+  public static ModelVersionMetaService getInstance() {
+    return INSTANCE;
+  }
+
+  private ModelVersionMetaService() {}
+
+  public List<ModelVersionEntity> listModelVersionsByNamespace(Namespace ns) {
+    NamespaceUtil.checkModelVersion(ns);
+
+    NameIdentifier modelIdent = NameIdentifier.of(ns.levels());
+    // Will throw a NoSuchEntityException if the model does not exist.
+    ModelEntity modelEntity = ModelMetaService.getInstance().getModelByIdentifier(modelIdent);
+
+    List<ModelVersionPO> modelVersionPOs =
+        SessionUtils.getWithoutCommit(
+            ModelVersionMetaMapper.class,
+            mapper -> mapper.listModelVersionMetasByModelId(modelEntity.id()));
+
+    if (modelVersionPOs.isEmpty()) {
+      return Collections.emptyList();
+    }
+
+    // Get the aliases for all the model versions.
+    List<ModelVersionAliasRelPO> aliasRelPOs =
+        SessionUtils.getWithoutCommit(
+            ModelVersionAliasRelMapper.class,
+            mapper -> mapper.selectModelVersionAliasRelByModelId(modelEntity.id()));
+    Multimap<Integer, ModelVersionAliasRelPO> aliasRelPOsByModelVersion =
+        ArrayListMultimap.create();
+    aliasRelPOs.forEach(r -> aliasRelPOsByModelVersion.put(r.getModelVersion(), r));
+
+    return modelVersionPOs.stream()
+        .map(
+            m -> {
+              List<ModelVersionAliasRelPO> versionAliasRelPOs =
+                  Lists.newArrayList(aliasRelPOsByModelVersion.get(m.getModelVersion()));
+              return POConverters.fromModelVersionPO(modelIdent, m, versionAliasRelPOs);
+            })
+        .collect(Collectors.toList());
+  }
+
+  public ModelVersionEntity getModelVersionByIdentifier(NameIdentifier ident) {
+    NameIdentifierUtil.checkModelVersion(ident);
+
+    NameIdentifier modelIdent = NameIdentifier.of(ident.namespace().levels());
+    // Will throw a NoSuchEntityException if the model does not exist.
+    ModelEntity modelEntity = ModelMetaService.getInstance().getModelByIdentifier(modelIdent);
+
+    Integer modelVersion;
+    try {
+      modelVersion = Integer.valueOf(ident.name());
+    } catch (NumberFormatException e) {
```

Comment:
It could use the `NumberUtils.isCreatable` to check if could convert to a number.